storage/database/shardedPersister.go

```diff
@@ -64,21 +64,33 @@ func (s *shardedPersister) computeID(key []byte) uint32 {
 
 // Put add the value to the (key, val) persistence medium
 func (s *shardedPersister) Put(key []byte, val []byte) error {
+	s.persistersMut.Lock()
+	defer s.persistersMut.Unlock()
+
 	return s.persisters[s.computeID(key)].Put(key, val)
 }
 
 // Get gets the value associated to the key
 func (s *shardedPersister) Get(key []byte) ([]byte, error) {
+	s.persistersMut.RLock()
```

Comment:
if you only add/remove/set in the persisters map inside constructor, you don't need a mutex

Reply:
done